error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/AssertJRules.java

```diff
@@ -264,6 +264,7 @@ void before(Iterable<E> iterable) {
       Refaster.anyOf(
           assertThat(iterable).hasSize(0),
           assertThat(iterable.iterator().hasNext()).isFalse(),
+          assertThat(iterable.iterator()).isExhausted(),
           assertThat(Iterables.size(iterable)).isEqualTo(0L),
           assertThat(Iterables.size(iterable)).isNotPositive());
     }
```

Comment:
The rules in this class are not yet tested, and require further cleanup/generalization. For now I just added this case here to keep track of it.